UI Upgrade

BryonLewis opened the pull request:
resolves #389

mvandenburgh commented on django/src/rdwatch/views/model_run.py:
```diff
@@ -469,10 +497,27 @@ def get_proposals_query(model_run_id: UUID4):
+        raise Http404()
+
+    query = get_sites_query(model_run_id)
+    model_run = data[0]
```
```suggestion
    model_run = data.first()
```

mvandenburgh commented on django/src/rdwatch/views/satellite_fetching.py:
```diff
@@ -0,0 +1,26 @@
+
+from rdwatch.models import SatelliteFetching
+
+router = Router()
+logger = logging.getLogger(__name__)
+
+
+@router.get('/running/')
+def satellite_fetching_running(
+    request: HttpRequest, model_runs: list[UUID4] = Query(None)  # noqa: B008
+):
+    running_sites = SatelliteFetching.objects.filter(
+        status=SatelliteFetching.Status.RUNNING
+    )
+    if model_runs is not None and len(model_runs) > 0:
+        running_sites = running_sites.select_related('sites').filter(
+            site__configuration__in=model_runs
+        )
+    running_site_ids = list(running_sites.values_list('site_id', flat=True))
+    return running_site_ids
```
```suggestion
router = RouterPaginated()
logger = logging.getLogger(__name__)


@router.get('/running/')
def satellite_fetching_running(
    request: HttpRequest, model_runs: list[UUID4] = Query(None)  # noqa: B008
):
    running_sites = SatelliteFetching.objects.filter(
        status=SatelliteFetching.Status.RUNNING
    )
    if model_runs is not None and len(model_runs) > 0:
        running_sites = running_sites.filter(
            site__configuration__in=model_runs
        )
    running_site_ids = running_sites.values_list('site_id', flat=True)
    return running_site_ids
```

Two things -
1) I don't think the select_related is necessary
2) Not paginating this endpoint + casting the full queryset to a list could result in excessive memory usage if there are thousands of satellite fetchings all at once. I don't think that would ever realistically happened outside of someone maliciously trying to crash the server, but paginating it is simple enough that it seems like it'd be worth it?

mvandenburgh commented on django/src/rdwatch_scoring/api.py:
```diff
@@ -12,6 +12,7 @@
 api.add_router('/regions/', views.region.router)
 api.add_router('/sites/', views.site.router)
 api.add_router('/eval-numbers/', views.eval.router)
+api.add_router('/satellite_fetching/', views.satellite_fetching.router)
```
```suggestion
api.add_router('/satellite-fetching/', views.satellite_fetching.router)
```
Same as above

mvandenburgh reviewed (changes requested):
Some preliminary comments/review on the backend code. Next, I'll take a pass at the frontend code, and then I'll pull the branch down and test it out in a final review

mvandenburgh commented:
I noticed something strange with the checkboxes on sites when testing this. Selecting a site by clicking it on the map doesn't fill the checkbox -
![Peek 2024-05-01 12-41](https://github.com/ResonantGeoData/RD-WATCH/assets/37340715/0554b180-d943-4a1e-9edb-a8d4918b8fe8)

I believe the checkbox should get filled here? Unless I'm misunderstanding the UX

BryonLewis commented:
> I believe the checkbox should get filled here? Unless I'm misunderstanding the UX

I had that on in the beginning and I don't remember why I turned it off.  Give me some time, I can easily turn it back on.

BryonLewis commented:
> I believe the checkbox should get filled here? Unless I'm misunderstanding the UX

I redid the selection logic for the regular RGD viewer and the annotation viewer.
I also fixed the scrolling tool.  It was working before I switched over to utilize virtual scroll lists.  Added code to handle this so control + hover over sites will now scroll to the site in the list that is being hovered over.

mvandenburgh commented:
Frontend code changes and the proposal view lgtm. Only thing I haven't reviewed/tested yet is the scoring app backend changes, my local scoring db got messed up so I have to repopulate that. I'll do that tomorrow and get a final review posted then.

mvandenburgh commented on django/src/rdwatch_scoring/views/satellite_fetching.py:
```diff
@@ -0,0 +1,24 @@
+import logging
+
+from ninja import Query, Router
+from pydantic import UUID4
+
+from django.http import HttpRequest
+
+from rdwatch_scoring.models import SatelliteFetching
+
+router = Router()
+logger = logging.getLogger(__name__)
+
+
+@router.get('/running/')
+def satellite_fetching_running(
+    request: HttpRequest, model_runs: list[UUID4] = Query(None)  # noqa: B008
+):
+    running_sites = SatelliteFetching.objects.filter(
+        status=SatelliteFetching.Status.RUNNING
+    )
+    if model_runs is not None and len(model_runs) > 0:
+        running_sites = running_sites.filter(model_run_uuid__in=model_runs)
+    running_site_ids = list(running_sites.values_list('site', flat=True))
+    return running_site_ids
```
This also needs to be updated as the `rdwatch` app endpoint was

BryonLewis replied:
I'm assuming you mean the Pagination and the response type?

mvandenburgh commented on vue/src/client/services/ApiService.ts:
```diff
@@ -162,6 +173,24 @@ export class ApiService {
+    ): CancelablePromise<SiteInfo> {
+      return __request(OpenAPI, {
+        method: "GET",
+        url: `${this.getApiPrefix()}/sites/{id}`,
```
```suggestion
        url: `${this.getApiPrefix()}/sites/{id}/`,
```

mvandenburgh commented on django/src/rdwatch/views/site_image.py:
```diff
@@ -23,7 +23,7 @@ class SiteImageSchema(Schema):
     cloudcover: float | None
     image: str
-    observation_id: UUID4 | None
+    observation_id: UUID4 | str | None
```
Leftover from debugging?
```suggestion
    observation_id: UUID4 | None
```

BryonLewis replied:
Yup

mvandenburgh commented on django/src/rdwatch_scoring/views/satellite_fetching.py:
```diff
@@ -0,0 +1,26 @@
+    )
+    if model_runs is not None and len(model_runs) > 0:
+        running_sites = running_sites.filter(model_run_uuid__in=model_runs)
+    running_site_ids = list(running_sites.values_list('site', flat=True))
```
```suggestion
    running_site_ids = running_sites.values_list('site', flat=True)
```

Coercing to a list negates the benefits of pagination here

BryonLewis replied:
scattterbrained today.

mvandenburgh reviewed (approved):
Just one last change needed to the scoring satellite fetch endpoint. Once that's addressed, I think this is good to go :+1:
